### PR TITLE
feat(invert): add invert

### PR DESCRIPTION
**Pull request by wondonghwi**
close: #116 

I added an invert function to the es-toolkit library. The invert function takes an object as input and returns a new object where the keys and values are swapped. This functionality is available in lodash but was not present in es-toolkit.

The invert function is particularly useful when you need to reverse the mapping of an object. For example, given an object with key-value pairs, the invert function will produce a new object where the original values become the keys and the original keys become the values.

스크린샷 2024-07-04 오전 1 45 37

For reference, you can see the lodash invert function [here.](https://lodash.com/docs/4.17.15#invert)

### CODE REVIEW

File: benchmarks/invert.bench.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+for (let i = 0; i < 10000; i++) {
+  object[`key${i}`] = `value${i}`;
+}
+
+describe('invert function benchmark', () => {
+  bench('Lodash invert', () => {
+    invertByLodash(object);
+  });
+
+  bench('Custom invert', () => {
```
Comment (hautest) [inline benchmarks/invert.bench.ts:15]: **custom** is good, but what do you think about doing it with `es-toolkit/invert` ?

Comment (end-jarry) [reply benchmarks/invert.bench.ts]: @hautest 
Thank you for the good comments, I've made the corrections more clearly!

File: src/object/invert.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @returns {{ [key in V]: K }} - A new object with keys and values inverted
+ *
+ * @example
+ * invert({ a: 1, b: 2, c: 3 }); // { 1: 'a', 2: 'b', 3: 'c' }
+ * invert({ 1: 'a', 2: 'b', 3: 'c' }); // { a: '1', b: '2', c: '3' }
+ * invert({ a: 1, 2: 'b', c: 3, 4: 'd' }); // { 1: 'a', b: '2', 3: 'c', d: '4' }
+ * invert({ a: Symbol('sym1'), b: Symbol('sym2') }); // { [Symbol('sym1')]: 'a', [Symbol('sym2')]: 'b' }
+ */
+
+export function invert<K extends string | number | symbol, V extends string | number | symbol>(
```
Comment (jgjgill) [inline src/object/invert.ts:20]: I think you can use `PropertyKey`.
They're the same, but I think it's more readable.

```suggestion
export function invert<K extends PropertyKey, V extends PropertyKey>(
```

Comment (end-jarry) [reply src/object/invert.ts]: @jgjgill 
I've made the modifications. Thank you for the good comments. I think it's more readable.

Comment (wondonghwi) [reply src/object/invert.ts]: @jgjgill 
I've made the modifications. Thank you for the good comments. I think it's more readable

File: benchmarks/invert.bench.ts (hunk 1 above)
Comment (wondonghwi) [reply benchmarks/invert.bench.ts]: @hautest 
Thank you for the good comments, I've made the corrections more clearly!

File: src/object/invert.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * invert({ a: 1, 2: 'b', c: 3, 4: 'd' }); // { 1: 'a', b: '2', 3: 'c', d: '4' }
+ * invert({ a: Symbol('sym1'), b: Symbol('sym2') }); // { [Symbol('sym1')]: 'a', [Symbol('sym2')]: 'b' }
+ */
+
+type PropertyKey = string | number | symbol;
+
+export function invert<K extends PropertyKey, V extends PropertyKey>(obj: Record<K, V>): { [key in V]: K } {
+  const result = {} as { [key in V]: K };
+
+  for (const key of Object.keys(obj) as K[]) {
```
Comment (raon0211) [inline src/object/invert.ts:25]:
```suggestion
  for (const key in obj) {
```

What about we simplify here using `for ... in` loop?

**Pull request merged**